projects/ngx-d3timeline/src/lib/store/drag-selectors.ts

```diff
@@ -0,0 +1,207 @@
+import { Orientation } from '../orientation';
+import { pointToTransform } from '../drag-util';
+import {
+  DraggedTimelineEvent,
+  draggingTimelineEvent
+} from './dragged-timeline-event';
+
+const selectInverseBandScale = createSelector(
+  [selectBandScale],
+  getInverseBandScale
+);
+
+const selectSeriesShiftedData = createSelector(
+  [selectData, selectInverseBandScale],
+  (data, inverseBandScale) =>
+    data.map(d => ({
+      ...d,
+      series: d.x !== null ? inverseBandScale(d.x) : d.series,
+      dx: 0
+    }))
+);
+
+export const foo = createSelector(
+  [selectDragEvent, selectSeriesShiftedData],
+  (
+    dragEvent: TimelineDragEvent,
+    data: DraggedTimelineEvent[]
+  ): DraggedTimelineEvent => dragEvent && data.find(d => d.id === dragEvent.id)
+);
+
+export const selectDraggingTimelineEvent = createSelector(
+  [foo, selectDragEvent],
+  (foo, dragEvent): DraggedTimelineEvent =>
+    foo && draggingTimelineEvent(foo, dragEvent)
+);
+
+const selectDeltaTime = createSelector(
+  [selectTimeOrientation, selectDragEvent],
+  (timeOrientation: Orientation, dragEvent: TimelineDragEvent) =>
+    timeOrientation === Orientation.Vertical
+      ? dragEvent && dragEvent.dy
+      : dragEvent && dragEvent.dx
+);
+
+const selectDeltaResource = createSelector(
+  [selectTimeOrientation, selectDragEvent],
+  (timeOrientation: Orientation, dragEvent: TimelineDragEvent) =>
+    timeOrientation === Orientation.Vertical
+      ? dragEvent && dragEvent.x
+      : dragEvent && dragEvent.y
+);
+
+const selectDropEventSeries = createSelector(
+  [selectInverseBandScale, selectDeltaResource],
+  (inverseBandScale: InverseBandScale, deltaResource: number) =>
+    inverseBandScale(deltaResource)
+);
+
+const selectShiftedTimesForDraggingTimelineEvent = createSelector(
+  [selectDraggingTimelineEvent, selectDeltaTime, selectTimeScale],
+  (
+    draggingTimelineEvent: TimelineEvent,
+    deltaTime: number,
+    timeScale: TimeScale
+  ) =>
+    draggingTimelineEvent && {
+      start: timeScale.invert(
+        timeScale(draggingTimelineEvent.start) + deltaTime
+      ),
+      finish: timeScale.invert(
+        timeScale(draggingTimelineEvent.finish) + deltaTime
+      )
+    }
+);
+
+const positionInTimeScale = createSelector(
+  [selectTimeScale],
+  (timeScale: TimeScale) => (timelineEvent: TimelineEvent) =>
+    timelineEvent && timeScale(timelineEvent.start)
+);
+
+const positionInBandScale = createSelector(
+  [selectBandScale],
+  (bandScale: BandScale) => (timelineEvent: TimelineEvent) =>
+    timelineEvent && bandScale(timelineEvent.series)
+);
+
+const getEventY = createSelector(
+  [selectTimeOrientation, positionInBandScale, positionInTimeScale],
+  (timeOrientation: Orientation, pt, pb) => (timelineEvent: TimelineEvent) =>
+    timeOrientation === Orientation.Vertical
+      ? pb(timelineEvent)
+      : pt(timelineEvent)
+);
+
+const getEventX = createSelector(
```

Comment:
> getEventX and getEventY could be getEventPosition? They have the same inputs, the position is the thing we need.

When we get the orientation strategy it will be straight forward to merge them into a getPositionSelector. Currently merging them would add a lot of branching logic in the getPositionSelector.